### PR TITLE
Gitea DevService Request and Org

### PR DESCRIPTION
**Pull request by iocanel**
The pull request adds the following:

- Support for creating a gitea organization on startup.
- Support for requesting the enablement of the Gitea DevService via build items.

### CODE REVIEW

File: deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java [hunk 1]
```diff
@@ -48,6 +48,12 @@ interface DevService {
         @WithDefault("quarkus")
         String adminPassword();
 
+        /**
+         * The organization to be created when the Dev Service starts.
+         */
+        @WithDefault("dev")
+        String organization();
```
Comment (gastaldi) [inline deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java:55]:
```suggestion
        Optional<List<String>> organizations();
```

File: deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java [hunk 2]
```diff
@@ -33,6 +36,14 @@ class GiteaContainer extends GenericContainer<GiteaContainer> {
         waitingFor(forListeningPorts(HTTP_PORT));
         // Needed for podman (see https://github.com/testcontainers/testcontainers-java/issues/7310)
         withStartupAttempts(2);
+
+        Optional<String> networkAlias = devServiceConfig.networkAlias()
+                .or(() -> devServiceRequest.map(GiteaDevServiceRequestBuildItem::getAlias));
+        networkAlias.ifPresent(alias -> {
+            withNetworkAliases(alias);
+            withNetwork(Network.SHARED);
+        });
```
Comment (gastaldi) [inline deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java:45]: This doesn't seem related to this PR?

File: deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java (hunk 1 above)
Comment (gastaldi) [reply deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java:55]: Actually instead of introducing a new org, what if the `repositories()` could specify them? Eg. `quarkus.jgit.devservices.repositories=dev/foo,bar` would create an org named `dev` and a repository `foo` in there and a `bar` in the user organization

Comment (gastaldi) [reply deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java]: This parameter is useless now that the organization name is included in the repository name

Comment (iocanel) [reply deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java]: Agree. Updated the pull request so that:
- user is able to specify org repositories using: <org>/<repo> pattern.
- user is able to specify user repositories using: <repo> pattern.
- user is able to specify organizations without repositories.

Comment (gastaldi) [reply deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java]: Ah, organizations without repositories, good call, haven't thought about that

Comment (gastaldi) [reply deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java]: Can you also update the `repositories()` Javadoc with this explanation?

File: deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java [hunk 3]
```diff
@@ -94,17 +164,81 @@ private void createRepository(GiteaContainer giteaContainer, String repository)
...
+                "-H",
+                "Content-Type: application/json",
+                httpUrl
+        };
+
+        log.debug(String.join(" ", cmd));
+        ExecResult execResult = execInContainer(cmd);
+
+        if (execResult.getExitCode() == 0) {
+            log.info("Organization exists: " + org);
```
Comment (gastaldi) [inline deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java:234]:
```suggestion
            log.debug("Organization exists: " + org);
```

File: deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java [hunk 4]
```diff
@@ -94,17 +164,81 @@ private void createRepository(GiteaContainer giteaContainer, String repository)
...
+            return true;
+        } else if (execResult.getExitCode() == 404) {
+            log.info("Organization does not exist: " + org);
```
Comment (gastaldi) [inline deployment/src/main/java/io/quarkus/jgit/deployment/GiteaContainer.java:237]:
```suggestion
            log.debug("Organization does not exist: " + org);
```

**Review by gastaldi — APPROVED** [review]
LGTM now, good job!
